FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift

```diff
@@ -74,8 +82,23 @@ final class EncryptedStorage: EncryptedStorageType {
 
 // MARK: - LogOut
 extension EncryptedStorage: LogOutHandler {
-    func logOut() { // todo - logOut is not clear - should be called onLogOut to make it clear it's responding to an event
-        destroyEncryptedStorage()
+    func logOutUser(email: String) throws {
+        let users = storage.objects(UserObject.self)
+
+        // in case there is only one user - just delete storage
+        if users.count == 1, users.first?.email == email {
+            destroyEncryptedStorage()
+        } else {
+            // remove user and keys for this user
+            let userToDelete = users.filter { $0.email == email }
+            let keys = storage.objects(KeyInfo.self).filter { $0.account == email }
+            let folders = storage.objects(FolderObject.self)
```

Comment:
Are folder rows also differentiated by users? Or are they getting shared among users?

Reply:
this is folders for an account, like inbox, snoozed, somePersonalFolderName

Reply:
How do you prevent folders from one account from mixing with folders from another account?

For example, for keys you do `.filter { $0.account == email }` but for folders you pull all. Or is there also an `account` column on the folders table?

Reply:
For now we do not associate folders with a user. 
We fetch and update folders in storage on every time user opens a side menu. 
I still need to associate this list of folders to a user.

Reply:
Understood

Reply:
Fixed in latest commit